src/ansi-c/expr2c.cpp

```diff
@@ -3794,6 +3794,77 @@ std::string expr2ct::convert_with_precedence(
   else if(src.id()==ID_type)
     return convert(src.type());
 
+  else if(src.id() == ID_rol)
+  {
+    // AAAA rol n ==   (AAAA << n % width(AAAA))
+    //               | (AAAA >> (width(AAAA) - n % width(AAAA)))
+    auto bin_expr = to_binary_expr(src);
+    // Get AAAA and if it's signed wrap it in a cast to remove
+    // the sign since this messes with C shifts
+    exprt &op0 = bin_expr.op0();
+    if(can_cast_type<signedbv_typet>(op0.type()))
+    {
+      // Shifts in C are arithmetic and care about sign, by forcing
+      // a cast to unsigned we force the shifts to perform rol/r
+      size_t ws = op0.type().get_size_t("width");
+      op0 = typecast_exprt(op0, unsignedbv_typet(ws));
+    }
+    // Get n because we need its type to construct the "width(AAAA)"
+    // value to use in operations below.
+    exprt &op1 = bin_expr.op1();
+    // Construct the "width(AAAA)" constant
+    irep_idt width_val = op0.type().get("width");
```

Comment:
:pick: Duplicated extraction of getting the width from `op0`. It should be possible to move the previous getting of the width out of the `if` statement and reuse it. We can expect the type to be a `bitvector_typet` whether or not it is signed. So extracting the width can be - `const size_t width = type_checked_cast<bitvector_typet>(op0.type()).get_width();`